frontend/src/components/NavBar.jsx

```diff
@@ -0,0 +1,38 @@
+// NavBar.js
+
+import React from 'react';
+import { Link, useMatch, useResolvedPath } from 'react-router-dom';
```

Comment:
None of these appear to be used. Fine to leave them if they have some future purpose.
Link is imported but navigate is used. I am not sure but have a feeling that you will be using wither one or the other.